plugins/cassandra/cassandraembed.cpp

```diff
@@ -311,20 +318,20 @@ class CassandraRetryingFuture : public CInterface
             limiter->wait();
         future = cass_session_execute(session, statement);
         if (limiter)
-            cass_future_set_callback(future, signaller, this); // Note - this will call the callback if the future has already completed
+            cass_future_set_callback(future, signaller, LINK(limiter)); // Note - this will call the callback if the future has already completed
     }
     static void signaller(CassFuture *future, void *data)
     {
-        CassandraRetryingFuture *self = (CassandraRetryingFuture *) data;
-        if (self && self->limiter)
-            self->limiter->signal();
+        LinkedSemaphore *sem = (LinkedSemaphore *) data;
+        sem->signal();
```

Comment:
(I hesitate to add a check for NULL, because it leads future readers to suppose that it might be possible for it to be NULL - which it is not)